src/04-spin-boson-model.md

```diff
@@ -0,0 +1,776 @@
+
+We'll use a Hamiltonian with a very large coupling of $U=1$ so that we'll be able to see things more clearly.
+
+```python
+H = 1*two_state + 1*bosons + 1*interaction
+```
+
+```python
+f, ax = hinton(H)
+ax.tick_params(axis='x',labelrotation=90)
+ax.set_title("Matrix elements of H     (Fig 4)");
+```
+
+The colour and size of the squares in Fig 4 give you a measure of the how large various matrix elements are. The off diagonal elements arise solely from the interaction part of the Hamiltonian - this is what allows one state to (in a sense) "mutate" into another.
+
+
+
+**Important detour on state numbers and labels**
+
+We'll study Fig 4 in more detail shortly, but for now I want to draw you attention to the labels for the rows and columns. For example:
+- $|3, 0 \rangle$ represents 3 bosons and a TSS state of |+>
+- $|3, 1 \rangle$ represents 3 bosons and a TSS state of |->
+
+It is hard to remember QuTiP's convention for the numbering of states. It is therefore very helpful to have a way to map these QuTiP number states to something more immediately recognisable - let's create such a map.
+
+```python
+possible_ns = range(0, max_bosons+1)
+possible_ms = ["+","-"]
+nm_list = [(n,m) for (n,m) in product(possible_ns, possible_ms)]
+```
+
+```python
+nm_list
+```
+
+Whenever we want to know what |n,±> state corresponds to QuTiP's number state e.g. 4, we can just do this:
+
+```python
+nm_list[4]
+```
+
+We can create some nice labels corresponding to the `nm_list`. This will make things plots easier to understand.
+
+```python
+def make_braket_labels(nm_list):
+    bra_labels = ["$\langle$"+str(n)+", "+str(m)+" |" for (n,m) in nm_list]
+    ket_labels = ["| "+str(n)+", "+str(m)+"$\\rangle$" for (n,m) in nm_list]
+    return bra_labels, ket_labels
+```
+
+```python
+bra_labels, ket_labels = make_braket_labels(nm_list)
+```
+
+We can now pass these labels into the Hinton diagram
+
+```python
+f, ax = hinton(H, xlabels=ket_labels, ylabels=bra_labels)
+ax.tick_params(axis='x',labelrotation=90,)
+ax.set_title("Matrix elements of H     (Fig 5)");
+```
+
+That's better!
+
+If we now take a closer look at the structure of the Hinton diagram we can see some interesting features when we follow a path that connects one state to another:
+
+```python
+print("                Matrix elements of H     (Fig 6)")
+Image(filename='./img/04-hinton.png') 
+```
+
+If we imagine starting a simulation with 0 bosons and the TSS in its + state, i.e. |0,+>, then Fig 6 suggests that:
+1. there are connections (albeit indirect) from |0,+> to many different states with many more bosons, e.g. $|0,+> \rightarrow |1,-> \rightarrow |2,+> \rightarrow |3,-> \rightarrow |4,+> ...$
+2. there are some states that are not accessible at all if we start in the |0,+> state
+
+
+On 1. These indirect connections provide a mechanism to achieve the down conversation that we saw hints of earlier (namely $|0,+> \rightarrow |3,->$).
```

Comment:
conversion   --- not conversation :)

General comment: we talk about down conversion because we already know that there will be rabi oscillation with a reasonable frequency. But in general one don't know it yet

Reply:
Indeed, we expect it but we haven't yet seen it. I will add "potential down conversion" to make clear that we are still not sure yet